src/impls/simdjson.rs

```diff
@@ -38,13 +40,14 @@ impl<'input, 'tape> OwnedDeser<'input, 'tape> {
         // Rust doens't optimize the normal loop away here
         // so we write our own avoiding the lenght
         // checks during push
-        let mut res = Vec::with_capacity(len);
-        unsafe {
-            res.set_len(len);
+        let res: Vec<OwnedValue> = unsafe {
+            let mut res: Vec<MaybeUninit<OwnedValue>> = Vec::with_capacity(len);
             for i in 0..len {
-                std::ptr::write(res.get_unchecked_mut(i), self.parse().unwrap());
+                res.get_unchecked_mut(i).write(self.parse().unwrap());
             }
-        }
+            res.set_len(len);
```

Comment:
@Licenser i am not 100% sure about the order here? Should we set the len before iterating?

Reply:
the original solution is the same one we have in simd-json that avoids most of the errors, I'm wondering if it is right, since the unwrap in parse() seems very dangerous.

perhas we just bite the pullet and push at this point and deal with errors? (I know the broken code is mine :joy:)

Reply:
I am not 100% sure what exactly you mean here.
Yes, the unwrap is dirty. Shall we return a Result and still keep the uninit hack? Maybe a small microbenchmark can verify that we dont add too much overhead?

Reply:
I admit I was still in simd-json-derive land the parse here shouldn't be able to fail since we already have a tape so unwrap is fine :)

The reason to set the length first is so that if something goes terribly wrong we don't have a situation where we have lingering objects in there. The `ptr::write` ensures that rust doesn't try to free the overwritten content.

Reply:
I did try to circumvent clippy complaining by using `MaybeUninit`, which was 2% slower than the original code, but then clippy suddenly stopped complainaing and i don't know why...